### ISSUE TITLE
Explain in docs how to import JSON files in Wasp on the server

### ISSUE THREAD

**Issue by Martinsos** [wasp-lang/wasp#1564]
Right now it is a bit tricky because you can directly import .json with node (at least 18), and people get surprised with it because error message is confusing (claims there is no file).

I explained how to go about it here: https://discord.com/channels/686873244791210014/1171780998699171841/1171780998699171841 .

We could add a small section in docs that covers this, since we had multiple people ask about this so far, even if it is not really Wasp-specific issue.

**Comment by Martinsos** [comment]
Another user asking about this: https://discord.com/channels/686873244791210014/1218225782183170158/1218225782183170158 .

**Comment by sodic** [comment]
Another one: https://discord.com/channels/686873244791210014/1318212644947230751/1318212644947230751

**Comment by Martinsos** [comment]
A bit of discussion we had on this:

@infomiho :

> If we start with the following structure:
> ```
> src
> ├── api.ts
> └──  static.txt
> ```
> and this is our API definition:
> ```ts
> ...
> export const getStaticResource = async (...) => {
>   const staticResource = fs.readFileSync("./src/static.txt", "utf8");
>   return res.json({ contents: staticResource });
> };
> ```
> We end up with ` Error: ENOENT: no such file or directory, open './src/static.txt'` because the API is executed from:  `.wasp/out/server/bundle/server.js`.
> 
> `process.cwd()` reports `/private/tmp/server-static/.wasp/out/server`
> 
> So... if we do this:
> ```ts
>   const cwd = process.cwd();
>   console.log("Current working directory", cwd);
> 
>   const fullPath = path.join(cwd, "../../../src/static.txt");
>   console.log("Full path", fullPath);
> ```
> We get `Full path /private/tmp/server-static/src/static.txt`. Which is correct and the thing works. _This is all in development._
> 
> Let's try building our server and running it: `wasp build` + `docker build . -t wasp-static-resource`.
> 
> Accessing the API, we get the following output:
> - `Current working directory /app/.wasp/build/server`
> - `Full path /app/src/static.txt`
> 
> And it's an error!
> 
> If I go into the container and check what do we have there... we only have:
> ```
> node_modules
> .wasp
> - build
> -- server
> --- bundle
> ---- server.js
> -- db
> 
> - out
> -- sdk
> --- wasp
> ---- ext-src
> ----- static.txt
> ```
> 
> So... we should be targeting the `.wasp/out/sdk/wasp/ext-src`! But this is kinda an internal details of Wasp (I believe <@772419159757422593> wants to change this in the future?)
> 
> We do this:
> ```ts
>   const cwd = process.cwd();
>   console.log("Current working directory", cwd);
> 
>   const fullPath = path.join(
>     cwd,
>     "../../../.wasp/out/sdk/wasp/ext-src/static.txt"
>   );
>   console.log("Full path", fullPath);
> ```
> - Works in dev ✅ 
> - Works when I run the server in a Docker container ✅
>
> TLDR: the user can do it - but he's relying on our internal SDK structure and something that we didn't intend for users to use. It's a hack that works now, but we it will probably break in the future for the user.

@Martinsos :

> Ok thanks for checking all this out @miho !
> 
> Is it correct if I summarize it like this:
> Server code, both in dev and prod, gets all bundled into .wasp/(out|build)/server/bundle/server.js .
> Static files that are not imported in the server code, like static.txt, are not part of that bundle. Instead, they are positioned somewhere else on the disk / in the Docker image, and that depends on dev/prod. This behaviour is not documented (where and if at all they will be there) and therefore accessing them currently relies on fragile relative paths.
> 
> Is this good?
> 
> If so, what are some solutions we have in mind?
> The only things that comes to my mind currently is to have a special static or assetsor something like that directory in which, whatever they put, we promise we will copy in an expected location (which they can obtain via either import or via ENV var). Kind of like cabal does with data_dir. Hah I used cabal as an example.

@infomiho :

> Static dir sounds like the way to go  
> 
> We already have public for the client (which might make this a bit confusing!) so I'm not sure where would we put this static dir - maybe we ask them to tell us the path in the Wasp file?

@Martinsos :

> What about server_data at top level?

**Comment by sodic** [comment]
Old one I forgot to link: https://discord.com/channels/686873244791210014/1278404045941375108/1278660290371649650

**Comment by sodic** [comment]
Another one: https://discord.com/channels/686873244791210014/1365788895010357349/1365788895010357349
